File: examples/app-redux-android-todolist-kotlin/src/debug/java/com/android/MainPresenter.java
package com.android;

import com.android.todolist.R;
import com.redux.AppAction;
import com.redux.DevToolPresenter;
import com.redux.AppState;

import javax.inject.Inject;

public class MainPresenter {
    private final DevToolPresenter<AppAction, AppState> devToolPresenter;

    @Inject
    public MainPresenter(DevToolPresenter<AppAction, AppState> devToolPresenter) {
        this.devToolPresenter = devToolPresenter;
    }

    public void bind(BaseActivity activity) {
        devToolPresenter.bind(activity.findViewById(R.id.dev_tools_drawer));
    }
}

File: examples/app-redux-android-todolist-kotlin/src/debug/kotlin/com/android/MainPresenter.kt
package com.android

import com.android.todolist.R
import com.redux.AppAction
import com.redux.AppState
import com.redux.DevToolPresenter
import javax.inject.Inject

class MainPresenter @Inject constructor(private val devToolPresenter: DevToolPresenter<AppAction, AppState>) {

    fun bind(activity: BaseActivity) = devToolPresenter.bind(activity.findViewById(R.id.dev_tools_drawer))
}
